src/utilities/utility.ts

```diff
@@ -47,6 +50,50 @@ export function checkForErrors(execResults, warnIfError?: boolean) {
     }
 }
 
+export function annotateChildPods(kubectl: Kubectl, resourceType: string, resourceName: string, allPods): IExecSyncResult[] {
+    const commandExecutionResults = [];
+    let owner = resourceName;
+    if (resourceType.toLowerCase().indexOf('deployment') > -1) {
+        owner = kubectl.getNewReplicaSet(resourceName);
+    }
+
+    if (allPods && allPods.items && allPods.items.length > 0) {
+        allPods.items.forEach((pod) => {
+            const owners = pod.metadata.ownerReferences;
+            if (owners) {
+                owners.forEach(ownerRef => {
+                    if (ownerRef.name === owner) {
+                        commandExecutionResults.push(kubectl.annotate('pod', pod.metadata.name, workflowAnnotations, true));
+                    }
+                });
+            }
+        });
+    }
+
+    return commandExecutionResults;
+}
+
+export function annotateNamespace(kubectl: Kubectl, namespaceName: string): IExecSyncResult {
+    const result = kubectl.getResource('namespace', namespaceName);
+    if (!result) {
+        return { code: -1, stderr: 'Failed to get resource' } as IExecSyncResult;

```

Comment:
Shouldn't the code be a non zero exit code and not a negative one?

Reply:
make it 1

Reply:
Done